src/Language/PureScript/CoreFn/Desugar.hs

```diff
@@ -221,11 +221,11 @@ findQualModules decls =
   fqValues :: A.Expr -> [ModuleName]
   fqValues (A.Var q) = getQual' q
   fqValues (A.Constructor q) = getQual' q
-  -- IsSymbol instances for literal symbols are automatically solved and the type
-  -- class dictionaries are built inline instead of having a named instance defined
-  -- and imported.  We therefore need to import the IsSymbol constructor from
-  -- Data.Symbol if it hasn't already been imported.
-  fqValues (A.TypeClassDictionaryConstructorApp C.IsSymbol _) = getQual' C.IsSymbol
+  -- Some instances are automatically solved (e.g IsSymbol, AppendSymbol, etc)
+  -- and the type class dictionaries are built inline instead of having a
+  -- named instance defined and imported.  We therefore need to import the type
+  -- class constructor from if it hasn't already been imported.
+  fqValues (A.TypeClassDictionaryConstructorApp typeClassName _) = getQual' typeClassName
```

Comment:
For Warn this won't change anything.  But for the other type classes that we solve for we should have been doing this.  But I can undo this for now and when I get more time tomorrow I'll make sure that there is an issue and make a new PR.